dcnm/resource_dcnm_inventory.go

```diff
@@ -538,20 +580,26 @@ func resourceDCNMInventroyUpdate(d *schema.ResourceData, m interface{}) error {
 
 			fabricID, err := extractFabricID(dcnmClient, fabricName)
 			if err != nil {
-				return err
+				return append(diags, diag.FromErr(err)...)
 			}
 
 			dUrl := fmt.Sprintf("/rest/control/fabrics/%s/inventory/test-reachability", strconv.Itoa(fabricID))
 			cont, err := dcnmClient.Save(dUrl, &inv)
 			if err != nil {
-				log.Println("error at test reachability for switch :", ip, err)
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Warning,
+					Detail:   fmt.Sprintf("error at test reachability for switch %s: %s", ip, err),
+				})
 				continue
 			}
 
 			switchM := extractSwitchinfo(cont)
 
 			if switchM.Selectable != "true" || switchM.Reachable != "true" {
-				log.Println("Desired switch: is not reachable or not selectable or invalid user/password or bad authentication protocol", ip)
+				diags = append(diags, diag.Diagnostic{
+					Severity: diag.Warning,
+					Detail:   fmt.Sprintf("Desired switch: is not reachable or not selectable or invalid user/password or bad authentication protocol %s", ip),
```

Comment:
probably should be "desired" based on other changes.

Reply:
Done